src/ui/components/HostPermissions/index.js

```diff
@@ -0,0 +1,151 @@
+/* @flow */
+import * as React from 'react';
+import { compose } from 'redux';
+
+import log from 'core/logger';
+import translate from 'core/i18n/translate';
+import type { I18nType } from 'core/types/i18n';
+import Permission from 'ui/components/Permission';
+
+/* eslint-disable no-continue */
+type Props = {|
+  i18n: I18nType,
+  permissions: Array<string>,
+|};
+
+export class HostPermissionsBase extends React.Component<Props> {
+  constructor(props: Props) {
+    const { i18n } = props;
+    super(props);
+    // These should be kept in sync with Firefox's strings for webextention
+    // host permissions which can be found in
+    // https://hg.mozilla.org/mozilla-central/raw-file/tip/browser/locales/en-US/chrome/browser/browser.properties
+    this.permissionStrings = {
```

Comment:
I would rather this be a function that returns strings based on arguments than an array that we access. This feels less idiomatic to me and harder to read elsewhere in the component because it looks like you're accessing data rather than calling a function.